_posts/2018-07-17-its-a-lot-more-snappy.md

```diff
@@ -0,0 +1,171 @@
+---
+layout: post
+title: "\"It's A Lot More Snappy\""
+date: 2018-07-17
+categories: PWA SPA JavaScript
+author: Rick Foxcroft
+image: img/offline_logo.png
+---
+
+## _‘A real world application of PWA technology’_
+
+Recently Mechanical Rock undertook the journey of building a progressive web app (PWA) for a client here in Perth. A PWA uses modern web capabilities to deliver a native app-like experience by providing offline support and resource caching. The application we were building needed to provide users the ability to raise issues on equipment in geographically isolated areas. Due to this isolation and unreliable networks, it was important that the app could still function while offline and store data, locally on devices. We decided to build a PWA so that previously fetched data could be stored locally on the device and the application could be used without a network connection and still serve the data to the user.
+
+Whilst building the app we encountered some performance issues. Over time the app would gradually become unresponsive to the point where it would take a few seconds to register a button click.
+
+## What Was The Problem?
+
+* The app had to deal with a large dataset. One of these datasets was an array of 205509 objects. Let’s refer to these objects as Pokemon.
+* The list of Pokemon had to be used in several places throughout the app and in some places not all of the data pertaining to each Pokemon was used.
+* A JSON file of this magnitude takes a fair amount of time to download. A simulation in chrome reveals that this may take up to 3 minutes to download on  a 1mb/s link.
```

Comment:
-`A simulation in chrome reveals that this`-

You don't need to clarify where you got the stat.

`This may take up to 3 minutes to download on  a 1mb/s link`